packages/rpc-types/package.json

```diff
@@ -92,5 +92,8 @@
                 "path": "./dist/index*.js"
             }
         ]
+    },
+    "dependencies": {
+        "@solana/codecs-strings": "workspace:2.0.0-development"
     }
```

Comment:
Hmmm this no longer makes the `@solana/rpc-types` package a dependency-free package which is a shame considering it is supposed to be this low-level bag of types that people can import from stress free of circular dependencies.

That being said, I'm okay with taking the stance that **codec packages are even lower-level packages** since they enable encoding and decoding any types.

I think this PR is a good change and therefore it's a good tradeoff but I just wanted to flag that so we make an explicit decision on this dependency tree.

Reply:
I feel similar sentiment @lorisleiva, although I don't have a hard stance on either. I think we should just iron out what this library and the types within it should do. Are they merely layouts, and thus should all be types with no dependencies? Or are they types *with assertions*?

Again on the topic of tree-shaking, if the assertion functions aren't used downstream, codecs should tree-shake out, right? If so it's prob not a big deal.

I do think codecs are probably our lowest-level considering their generic use and lack of dependencies on other packages in our library.

If we went the layout route, where would it make sense for `Blockhash` to live @mcintyre94?

Reply:
Yeah they would all tree-shake anyway. I'm more concerned about creating an explicit pattern with our package dependencies. Breaking things up in micro-modules is now but comes at the price of being careful with how we manage these dependencies.

My stance on this is:
- **Codecs are the lowest level packages of the whole library**. We should see this as low-level as using a `Uint8Array`. Similarly to how legacy web3.js would use borsh layouts or bs58. That means, codecs should never start importing stuff like `@solana/transaction`. Never!
- **Type packages** such as `rpc-types` should be **repositories of re-usable types with the helper functions** attached to them. Think of them like tiny type modules. We've even got a nice convention there for these helpers — e.g. `MyType`, `getMyTypeEncoder()`, `getMyTypeDecoder()`, `getMyTypeCodec()`, `isMyType()`, `assertIsMyType()` and even `myType()` to create such type when applicable.
- Everything else is just packages built on top of that in a careful hierarchy to avoid circular dependencies, creating more dedicated type packages when required.

P.S.: just pasting the current web3.js dependency graph here as well so it's publicly accessible.

![Web3 js dependency graph](https://github.com/solana-labs/solana-web3.js/assets/3642397/450ca7dc-5f37-4a74-ad68-465500491ade)

Reply:
Yep, I agree with this approach!

Maybe before we add types to `@solana/rpc-parsed-types`, we hash out the orange arrows in your diagram and follow the above proposed plan?

Reply:
I had the same thought about adding a dependency here.

The other option here is to leave `assertIsBlockhash` with `transactions`, or move it somewhere else. But `rpc-types` does have the other assert functions colocated with the types it defines, and I thought that consistency was more important than avoiding a codecs dependency.

IMO this colocation makes the most sense. I agree with Loris' comment above that these packages should include their helper functions in general. I don't really mind thought because if we move them somewhere else and just make sure it's exported from the main library, it'll be indistinguishable to most devs. We should be consistent though.

Reply:
@mcintyre94 Agree with everything. I don't even think `rpc-parsed-types` having actual logic is an issue as long as its dependencies can only be at most: `codecs` and `rpc-types`.

Just let me know when you're happy with me detangling these types from `rpc-core` and `rpc-transport` because I expect if I do that now, we'll have lots of annoying merge conflicts to deal with haha.

Reply:
@lorisleiva I think I can probably finish these parsed account types today

I think the best thing is probably if I keep doing the minimum changes necessary for these (basically move stuff these need to rpc-types), and then you take over with the wider changes after that? I don't think anything I do will add work to that refactor!

Reply:
Perfect plan! 🙇‍♂️